README.md

```diff
@@ -2,20 +2,104 @@
 
 > Test client for Xo-Server
 
-## Usage
+You should have a `admin@admin.net` account with password `admin`.
 
-xo-server needs to be installed on `localhost:9000`.
-You should also have a `admin@admin.net` account with password `admin`.
+## Adding a test
 
-There should be a Xen Server available at `192.168.1.3`with the credentials `root` and password `qwerty`.
+### Organization
 
-Warning ! This client will create and delete a lot of items (users, groups, servers ...) to test features
+```
+src
+├─ user
+|   ├─ __snapshots__
+|   |     └─ index.spec.js.snap 
+|   └─ index.spec.js
+├─ job
+¦   └─ index.spec.js
+¦
+¦
+├─ _xoWithTestHelpers.js
+└─ util.js
+```
 
+The tests can describe xo methods or scenarios:  
 ```
-> npm ci
-> npm test
+describe("user", () => {
+  // testing a method
+  describe(".create(): ", () => {
+    it("creates a user with permission", () => {
+      /* some tests */
+    });
+  });
+      
+  // testing a scenario
+  if("creates two users, modifies a user email to be the same with the other and fails trying to connect them", () => {
+    /* some tests */
+  });
+});
 ```
 
+### Test environment
+
+- The test environment must remain the same before and after each test:  
+  * each resource created must be deleted  
+  * existing resources should not be altered
+
+### Best practices
+
+- Make a sentence for the title of the test. It must be clear and consistent.
+
+- If the feature you want to test is not implemented : write it and skip it, using `it.skip()`.
+
+- Take values ​​that cover the maximum of testing possibilities.
+
+- If you make tests which keep track of large object, it is better to use snapshots.  
+
+- There is a helper `_xoWithTestHelpers.js` to create temporary resources and to interface with XO.  
+  You can use it if you need to create resources which will be automatically deleted after the test:  
+  ```
+  describe(".create() :", () => {
+    it("creates a user without permission", async () => {
+      // The user will be deleted automatically at the end of the test
+      const userId = await xo.createTempUser({
+        email: "wayne1@vates.fr",
+        password: "batman1",
+      });
+      expect(omit(await xo.getUser(userId), "id")).toMatchSnapshot();
```

Comment:
Where does `xo` come from?

Reply:
Adding an `import xo from "util"` at the begin of this code is enough ?

Reply:
I don't know, but a user reading the documentation should know how to write a simple test from this doc only.